airbyte-api/src/main/openapi/config.yaml

```diff
@@ -4564,6 +4634,10 @@ components:
           $ref: "#/components/schemas/AirbyteCatalog"
         schedule:
           $ref: "#/components/schemas/ConnectionSchedule"
+        scheduleType:
```

Comment:
Why is it not encapsulated in one object?
```
Schedule:
  type:
  data:
```

Reply:
In my opinion the deeper nesting would make it a bit more unwieldy without _too_ much benefit - encapsulating it under an object doesn't really reduce the cognitive load since every user has to figure out what to do with schedule data anyway whether it's top-level or nested.

I did consider the approach you propose, and I'm not really opposed to it. It comes with the wrinkle that during the migration we have this schedule object that contains both the old and new schemas. (Or - we have some transition period where we have the new schema with some new name, remove the old schema, and then migrate again the new schema to the existing `Schedule` name.) All in all this felt simpler; but open to feedback here.

Reply:
No strong opinion on my end. However everything related to migration complexity is a non-issue to me since we are v0. I prefer to have something clean that we want than to make design tradoffs based on backward comp or migrations.